src/Tizen.Applications.Common/Tizen.Applications/AppControl.cs

```diff
@@ -444,6 +445,48 @@ public ExtraDataCollection ExtraData
             }
         }
 
+        /// <summary>
+        /// Gets and sets the component ID.
+        /// </summary>
+        /// <value>
+        /// (if the component ID is null for setter, it clears the previous value.)
+        /// </value>
+        /// <example>
+        /// <code>
+        /// AppControl appControl = new AppControl();
+        /// appControl.ComponentId = "org.tizen.frame-component";
+        /// Log.Debug(LogTag, "ComponentId: " + appControl.ComponentId);
+        /// </code>
+        /// </example>
+        /// <since_tizen> 6 </since_tizen>
+        public string ComponentId
```

Comment:
What is the role of this property?
How is it related to ApplicationId?
Why don't you extend AppControl to ComponentControl?

Reply:
To send a send launch request to execute a component of the component-based-application, the component ID is necessary.
To execute the component properly, the AppControl should have the application ID and the component ID.

Reply:
Is Component ID not globally unique?

Reply:
How about extend?
``` c#
 class ComponentControl : AppControl 
```

Reply:
Component ID is unique globally.
It's an additional property of AppControl.
If the component ID doesn't exist, the application(component-based-application) is launched.

Reply:
> To execute the component properly, the AppControl should have the application ID and the component ID.

I can't understand why need appid? you said  `Component ID is unique globally`

Reply:
As I mentioned, the component Id is the additional property. If it does not exist on AppControl, the application is launched. If it exists, the component is launched.
Setting the component Id is not mandatory.

Reply:
The component ID is not required to launch ComponetBasedApplication. If it is not added, the main component will be launched.
If we provide extended ComponentControl, developer may be confused about which class to use when launching only with the App ID without the component ID.
In fact, the caller app may not know that the target app is a ComponentBasedApplication.

Reply:
Ok, I understand, but ComponentId is not proper to AppControl's property, because it meansingless in AppControl object in most cases.

Reply:
If we provide it as property, developer may be confused "What is it and which value should be set?"